### ISSUE TITLE
Move threading.local docs from docstring to the docs

### ISSUE THREAD

**Issue by nedbat** [python/cpython#107006]
# Documentation

https://docs.python.org/3/library/threading.html#threading.local says: 

> For more details and extensive examples, see the documentation string of the _threading_local module: [Lib/_threading_local.py](https://github.com/python/cpython/tree/3.11/Lib/_threading_local.py).

If we have good helpful words about a component in the standard library, we should publish them in the docs.  The docstring should be moved to the .rst file, or at least parts harvested.

### Linked PRs
* gh-131840
* gh-133432

**Comment by mdickinson** [comment]
Yes, please! It would be particularly useful to have the pattern of subclassing `threading.local` available in the docs.

**Comment by hugovk** [comment]
Thanks all, this is now done.
